service/history/replication/executable_task.go

```diff
@@ -316,14 +332,37 @@ func (e *ExecutableTaskImpl) Resend(
 			tag.WorkflowRunID(retryErr.RunId),
 		)
 		// workflow is not found in source cluster, cleanup workflow in target cluster
-		return e.DeleteWorkflow(
+		return false, e.DeleteWorkflow(
 			ctx,
 			definition.NewWorkflowKey(
 				retryErr.NamespaceId,
 				retryErr.WorkflowId,
 				retryErr.RunId,
 			),
 		)
+	case *serviceerrors.RetryReplication:
+		// it is possible that resend will trigger another resend, e.g.
+		// 1. replicating a workflow which is a reset workflow (call this workflow `new workflow`)
+		// 2. base workflow (call this workflow `old workflow`) of reset workflow is deleted on
+		//	src cluster and never replicated to target cluster
+		// 3. when any of events of the new workflow arrive at target cluster
+		//  a. using base workflow info to resend until branching point between old & new workflow
+		//  b. attempting to use old workflow history events to replay for mutable state then apply new workflow events
+		//  c. attempt failed due to old workflow does not exist
+		//  d. return error to resend new workflow before the branching point
+
+		// handle 2nd resend error, then 1st resend error
+		if _, err := e.Resend(ctx, remoteCluster, resendErr, remainingAttempt); err == nil {
```

Comment:
Should we handle it in NDCHistoryResender?

Reply:
there is another functionality which handles this case:
T = 0: received a replication task
T = 1: resend due to missing some events
T = 2: workflow deleted (due to retention) from source cluster
T = 3: above replication task & corresponding workflow should be deleted

we probably should (later) move these special case handling into a dedicated place